pkg/webhook/resources/templateversion/validator.go

```diff
@@ -75,6 +78,19 @@ func (v *templateVersionValidator) Create(_ *types.Request, newObj runtime.Objec
 		}
 	}
 
+	// Check JSON data in the annotations. This must be valid JSON data, as
+	// otherwise the IndexFunc of the cache couldn't process the VMTemplateVersion
+	volumeClaimTemplateString, ok := vmTemplVersion.Spec.VM.ObjectMeta.Annotations[util.AnnotationVolumeClaimTemplates]
+	if ok && volumeClaimTemplateString != "" {
+		var volumeClaimTemplates []corev1.PersistentVolumeClaim
+		if err := json.Unmarshal([]byte(volumeClaimTemplateString), &volumeClaimTemplates); err != nil {
```

Comment:
another question:

The following `func (v *templateVersionValidator) Update` makes sure the object can't be changed manually; if the templateString is only added/updated by the controller, how could the JSON is buggy?